DOCS/man/options.rst

```diff
@@ -1410,6 +1410,10 @@ Audio
     If the channel layout of the media file (i.e. the decoder) and the AO's
     channel layout don't match, mpv will attempt to insert a conversion filter.
 
+    The ``wasapi`` AO's channel layout is stereo in shared mode.
+    ``--audio-exclusive=yes`` is required for direct hardware output of any
```

Comment:
I'm not sure this is generally true. I assume you've specified `--audio-channels=auto` rather than the default `--audio-channels=auto-safe` which does indeed do as you describe. Other than that, it's up to Windows / drivers to decide whether it will accept multi-channel in shared mode.

Reply:
I played around with Windows audio settings a bit, turns out in shared mode `--audio-channels=auto` still upmixes/downmixes to the system's preferred channel layout set in Control Panel, whether it's stereo, quadraphonic, 5.1, etc.

Should I move this edit under `--audio-channels=auto` and rewrite it to state that `auto` acts like `auto-safe` with the `wasapi` AO in shared mode, or remove it entirely?

The reason I want to include a note like this is because I spent a good hour diagnosing why multichannel PCM bitstreaming wasn't working (Google didn't turn up much on this topic, especially on Windows), and figured this information could help someone else down the road.

Reply:
Actually looking at the code more closely, maybe `auto` / `auto-safe` doesn't affect wasapi shared mode at all. It probably really is up to Windows / drivers.

The thing is that shared mode requires all audio producing applications to be mixed into a single stream sent to the sound card (or over HDMI). Obviously this can't change just because one application decides it wants to output in a different format.

Some drivers may offer the possibility to accept different channel layouts, and then do mixing themselves rather than requiring mpv to do it, but it has to happen somewhere if the media's channel layout doesn't match the system-wide shared mode format.

Anyway, wouldn't you just set the system shared mode channel layout to how your speakers are actually set up and have everything mixed to that?

It might be worthwhile clarifying for `audio-channels=auto` that you may need to tweak your "system's preferred channel layout" to get multi-channel layout as it may be set to stereo by default.

Reply:
The intention is that `auto-safe` uses the system multichannel layout as advertised, but doesn't use it for direct hardware access, such as ALSA or wasapi exclusive mode. The reason is that sending unsupported channel layouts to connected hardware will result in noise or discarded channels or otherwise wrong playback. This can actually happen with HDMI.